crates/league-toolkit/src/core/wad/builder.rs

```diff
@@ -177,9 +177,17 @@ impl WadBuilder {
                 encoder.read_to_end(&mut compressed_data)?;
             }
             WadChunkCompression::Zstd => {
-                let mut encoder = zstd::Encoder::new(BufWriter::new(&mut compressed_data), 3)?;
-                encoder.write_all(data)?;
-                encoder.finish()?;
+                #[cfg(feature = "zstd")]
+                {
+                    let mut encoder = zstd::Encoder::new(BufWriter::new(&mut compressed_data), 3)?;
+                    encoder.write_all(data)?;
+                    encoder.finish()?;
+                }
+                #[cfg(feature = "ruzstd")]
+                {
+                    // TODO: impl ruzstd compress for wad
+                    return Err(WadBuilderError::UnsupportedCompressionType(compression));
```

Comment:
We should probably just panic here instead.

Reply:
I've just implemented it instead